packages/fuels-accounts/src/provider.rs

```diff
@@ -622,22 +620,17 @@ impl Provider {
 
     pub async fn get_message_proof(
         &self,
-        tx_id: &str,
-        message_id: &str,
-        commit_block_id: Option<&str>,
+        tx_id: &TxId,
+        message_id: &MessageId,
+        commit_block_id: Option<&BlockId>,
```

Comment:
I understand that it could be similar to `BlockHeight` but here maybe we want to make sure the user doesn't mistake a `BlockId` for  a `MessageId` and so on?

Reply:
Fair enough, ok by me. Don't forget to add the exports to `fuels`.

Reply:
lol I pushed to the wrong branch

Reply:
I'd either add the `BlockId` to the reexports (even though it's an alias for `Bytes32`) or would change `BlockId` in `provider.rs` to `Bytes32`. 

So that users don't have to see the declaration of `get_message_proof` and see that they need a `BlockId` and proceed to make one only to get frustrated that their LSP isn't giving any suggestions for it.

Reply:
Figured that `BlockId` is not the same level of "types" as the other like `MessageId` and decided to go for replacing it with `Bytes32`.